### PR TITLE
[TIR] compact buffer region

### PR DESCRIPTION
**Pull request by spectrometerHBH**
The compacted buffer's shape should be at most the same as the original shape.
cc @junrushao1994

### CODE REVIEW

**Comment by junrushao1994** [comment]
Thanks for the fix! It's essential for AutoTIR performance!! Please add a regression test :-)

**Comment by junrushao1994** [comment]
@spectrometerHBH would you like to rebase? Looks like the CI is very flaky

File: src/tir/transforms/compact_buffer_region.cc [hunk 1]
```diff
@@ -53,12 +53,30 @@ Region SimplifyAndNarrowBufferRegionFromNDIntSet(const NDIntSet& nd_int_set,
   for (size_t i = 0; i < nd_int_set.size(); ++i) {
     const arith::IntSet& int_set = nd_int_set[i];
     Range range = int_set.CoverRange(Range(/*begin=*/0, /*end=*/original_shape[i]));
-    result.push_back(
-        Range::FromMinExtent(analyzer->Simplify(range->min), analyzer->Simplify(range->extent)));
+    result.push_back(Range::FromMinExtent(
+        analyzer->Simplify(range->min), analyzer->Simplify(min(original_shape[i], range->extent))));
```
[the remark this answers is not in the document]
Comment (spectrometerHBH) [reply src/tir/transforms/compact_buffer_region.cc:57]: We can do.

**Comment by junrushao1994** [comment]
@spectrometerHBH we need to get this PR in for performance alignment

**Review by junrushao1994 — APPROVED** [review]

**Review by Hzfengsy — APPROVED** [review]

**Pull request merged**